client/rest/src/main/java/org/opensearch/client/StreamingRequest.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.client;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import org.reactivestreams.Publisher;
+
+import static java.util.Collections.unmodifiableMap;
+
+/**
+ * HTTP Streaming Request to OpenSearch. <strong>This is an experimental API.</strong>
+ */
+public class StreamingRequest<T> {
+    private final String method;
+    private final String endpoint;
+    private final Map<String, String> parameters = new HashMap<>();
+
+    private RequestOptions options = RequestOptions.DEFAULT;
```

Comment:
Can be final, I think, though it looks like this is never changed from the default. Is that right?

Reply:
No, we actually do change it (recently introduced)